quantum/pointing_device_drivers.c

```diff
@@ -109,17 +112,34 @@ report_mouse_t cirque_pinnacle_get_report(report_mouse_t mouse_report) {
     mouse_xy_report_t report_x = 0, report_y = 0;
     static bool       is_z_down = false;
 
-    cirque_pinnacle_scale_data(&touchData, cirque_pinnacle_get_scale(), cirque_pinnacle_get_scale()); // Scale coordinates to arbitrary X, Y resolution
+#    if !CIRQUE_PINNACLE_POSITION_MODE
+#        error Cirque Pinnacle with relative mode not implemented yet.
+#    endif
+
+    if (!touchData.valid) {
+        return mouse_report;
+    }
+
+#    if CONSOLE_ENABLE
+    if (debug_mouse && touchData.touchDown) {
+        dprintf("cirque_pinnacle touchData x=%4d y=%4d z=%2d\n", touchData.xValue, touchData.yValue, touchData.zValue);
+    }
+#    endif
+
+    // Scale coordinates to arbitrary X, Y resolution
+    cirque_pinnacle_scale_data(&touchData, cirque_pinnacle_get_scale(), cirque_pinnacle_get_scale());
 
     if (x && y && touchData.xValue && touchData.yValue) {
-        report_x = (mouse_xy_report_t)(touchData.xValue - x);
-        report_y = (mouse_xy_report_t)(touchData.yValue - y);
+        report_x = (int8_t)(touchData.xValue - x);
+        report_y = (int8_t)(touchData.yValue - y);
```

Comment:
```suggestion
        report_x = (mouse_xy_report_t)(touchData.xValue - x);
        report_y = (mouse_xy_report_t)(touchData.yValue - y);
```
Updated since #16371

Reply:
Done - thanks!